File: 206.reverse-linked-list.py
#
# @lc app=leetcode id=206 lang=python3
#
# [206] Reverse Linked List
#

# @lc code=start
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        newHead = ListNode()

        current = head
        while current != None:
            tempNext = newHead.next
            newHead.next = current
            
            current = current.next
            newHead.next.next = tempNext
        
        return newHead.next

# @lc code=end

File: 206.reverse-linked-list.js
/*
 * @lc app=leetcode id=206 lang=javascript
 *
 * [206] Reverse Linked List
 */

// Definition for singly-linked list.
function ListNode(val, next) {
  this.next = (next===undefined ? null : next)
  this.val = (val===undefined ? 0 : val)
}
// @lc code=start
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
  var newHead = new ListNode();
  while (head !== null) {
    var currentNode = head;
    head = head.next;
    
    var tempNode = newHead.next;
    newHead.next = currentNode;
    newHead.next.next = tempNode;
    
  }

  return newHead.next;
};
// @lc code=end

